src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -255,13 +255,31 @@ module GenericSimple = {
     | `Uniform({high}) => high
     | `Float(n) => n;
 
+
+  /* This function returns a list of x's at which to evaluate the overall distribution (for rendering).
+     This function is called separately for each individual distribution.
+
+     When called with xSelection=`Linear, this function will return (sampleCount) x's, evenly
+     distributed between the min and max of the distribution (whatever those are defined to be above).
+
+     When called with xSelection=`ByWeight, this function will distribute the x's such as to
+     match the cumulative shape of the distribution. This is slower but may give better results.
+  */
   let interpolateXs =
       (~xSelection: [ | `Linear | `ByWeight]=`Linear, dist: dist, sampleCount) => {
+
     switch (xSelection) {
     | `Linear => E.A.Floats.range(min(dist), max(dist), sampleCount)
     | `ByWeight =>
-      E.A.Floats.range(minCdfValue, maxCdfValue, sampleCount)
-      |> E.A.fmap(x => inv(x, dist))
+      switch (dist) {
+      // In `ByWeight mode, uniform distributions get special treatment because we need two x's
+      // on either side for proper rendering (just left and right of the discontinuities).
+      | `Uniform(n) => E.A.of_list([n.low -. minCdfValue, n.low +. minCdfValue,
+                                    n.high -. minCdfValue, n.high +. minCdfValue])
```

Comment:
A few things here:
1) It's easy to just make this an array in ReasonML. All you need is different brackets.
```
[|n.low -. minCdfValue, n.low +. minCdfValue,
                                    n.high -. minCdfValue, n.high +. minCdfValue|]
```
2) minCdfValue is meant to represent a percentile. .0001 tells us to use the 0.0001 percentile as the lower value for CDFs. It's use here seems very different, like it's being used as an absolute value that doesn't change no matter the scale. If the user wrote ``uniform(1M, 100M)``, it would use intervals of 0.0001, as it would with ``uniform(0.000001, 0.000002)``. It's not obvious what an amazing solution would be. I assume at least it should depend on the width of the uniform. Maybe something like 0.00001 of whatever the width is? Ideally it wouldn't need a different x value and we could handle shapes with multiple coordinates at the same x point, but this could be outside the scope here. If the code for uniforms is a bit difficult, it may be better to put it in the ``Uniform`` module above, but I'm not sure what the best practice is in functional programming

Reply:
Good point, it's fair to say that I'm abusing minCdfValue here and I hadn't considered tiny intervals. I think your suggestion of just using 0.00001 * width makes sense. I had tried moving the calculation into the `Uniform` module, but that just felt like an extra layer of spaghetti for no real benefit to the reader, so I kept it in the function

Reply:
Also, thanks for your very valid syntax/style suggestions. Please bear with me as I'm slowly getting used to Reason again :)